app/src/main/java/pl/droidevs/books/library/BookActivity.java

```diff
@@ -0,0 +1,115 @@
+package pl.droidevs.books.library;
+
+import android.app.Activity;
+import android.content.Loader;
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.os.AsyncTask;
+
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.res.ResourcesCompat;
+import android.support.v4.graphics.ColorUtils;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.graphics.Palette;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.ImageView;
+
+import com.bumptech.glide.Glide;
+
+
+import net.opacapp.multilinecollapsingtoolbar.CollapsingToolbarLayout;
+
+import java.security.PKCS12Attribute;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import pl.droidevs.books.R;
+
+public class BookActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        getSupportActionBar().hide();
+        setContentView(R.layout.activity_book);
+
+        ImageView imageView = (ImageView) findViewById(R.id.albumImageView);
+        Glide.with(this).load(R.drawable.orient_express).into(imageView);
+        CollapsingToolbarLayout collapsingToolbarLayout = (CollapsingToolbarLayout) findViewById(R.id.collapsingToolbarLayout);
+
+        Resources reso = this.getResources();
+        Bitmap image = BitmapFactory.decodeResource(reso, R.drawable.orient_express);
+
+        Palette.Swatch swatch = getDominantColor(image);
+        if (swatch != null) {
+            float[] hsl = swatch.getHsl();
+            float light = hsl[2];
+            float light2 = light > 0.075f ? light - 0.075f : 0f;
+            float[] hsl2 = {hsl[0], hsl[1], light2};
+            int color = ColorUtils.HSLToColor(hsl2);
+
+            collapsingToolbarLayout.setContentScrimColor(swatch.getRgb());
+            Window window = getWindow();
+            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+            window.setStatusBarColor(color);
+        }
+    }
+
+    public Palette.Swatch getDominantColor(Bitmap bitmap) {
+        List<Palette.Swatch> swatchesTemp = Palette.from(bitmap).generate().getSwatches();
+        List<Palette.Swatch> swatches = new ArrayList<Palette.Swatch>(swatchesTemp);
+        Collections.sort(swatches, new Comparator<Palette.Swatch>() {
+            @Override
+            public int compare(Palette.Swatch swatch1, Palette.Swatch swatch2) {
+                return swatch2.getPopulation() - swatch1.getPopulation();
+            }
+        });
+
+
+        return swatches.size() > 0 ? swatches.get(0) : null;
+    }
+
+    public class AverageRGBLoader extends AsyncTask<Bitmap, Void, Integer> {
```

Comment:
Unikałbym AsyncTasków - prowadzą do memory leaków. Co się stanie, jak system odtworzy aktywność po np. zmianie orientacji?

Reply:
Wydawało mi się, że AsyncTask to jedno z lepszych wyjść. A akurat ten się chyba i tak nie przyda.